File: app/src/main/java/com/example/skyeos/SkyeOsApplication.java
package com.example.skyeos;

import android.app.Application;

public class SkyeOsApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        AppLocaleManager.applyStoredLocale(this);
        AppGraph.getInstance(this).database.warmUp();
    }
}

File: app/src/main/java/com/example/skyeos/SkyeOsApplication.kt
package com.example.skyeos

import android.app.Application
import com.example.skyeos.data.db.LifeOsDatabase
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

@HiltAndroidApp
class SkyeOsApplication : Application() {

    @Inject
    lateinit var database: LifeOsDatabase

    override fun onCreate() {
        super.onCreate()
        AppLocaleManager.applyStoredLocale(this)
        database.warmUp()
    }
}
